lib/std/math/big/int.zig

```diff
@@ -1959,6 +1959,45 @@ pub const Const = struct {
         return bits;
     }
 
+    /// @popCount with two's complement semantics.
+    /// Asserts that bit_count is enough to represent value in two's compliment
+    /// and that the final count fits in a usize.
+    pub fn popCount(self: Const, bit_count: usize) usize {
+        if (self.eqZero()) return 0;
+
+        var sum: usize = 0;
+        if (self.positive) {
+            for (self.limbs) |limb| {
+                sum += @popCount(limb);
+            }
+        } else {
+            assert(self.fitsInTwosComp(if (self.positive) .unsigned else .signed, bit_count));
+
+            var remaining_bits = bit_count;
+            var carry: u1 = 1;
+            var add_res: Limb = undefined;
+
+            // All but the most significant limb.
+            for (self.limbs[0 .. self.limbs.len - 1]) |limb| {
+                carry = @boolToInt(@addWithOverflow(Limb, ~limb, carry, &add_res));
+                sum += @popCount(add_res);
+                remaining_bits -= limb_bits; // Asserted not to undeflow by fitsInTwosComp
+            }
+
+            // The most significant limb may have fewer than @bitSizeOf(Limb) meaningful bits,
+            // which we can detect with @clz().
+            // There may also be fewer limbs than needed to fill bit_count.
+            const limb = self.limbs[self.limbs.len - 1];
+            const leading_zeroes = @clz(limb);
+            assert(limb != 0);
+            sum += @popCount(~limb + carry); // Will never overflow because ~limb cannot be all 1s
```

Comment:
~limb = 1..1 for limb=0, but this does not explain why limb=0 is impossible at this point.
This comment feels also more like the assertion should be briefly explained.

Reply:
There is an assertion on the line above, but it could be better explained -- I'll give that a shot.

Reply:
Oh, I think I see why this is not obvious.

This assumes there are no extraneous limbs (ones with all zeroes) at the Most Significant end. I believe this is always true when setting values with Managed/Mutable.

If someone manually fills limbs with data, this may not hold. Is this an expected usage of Const bigints?

Reply:
Just checked, and there seem to be many places in this file that assume that a) `positive` is true for the zero value, and b) there are no trailing empty limbs. So while it would be easy to handle, I think it makes sense to to follow that assumption.

I made an assertion and added a note in the doc comment.